test/integration/test_retrieve_job.py

```diff
@@ -150,6 +150,21 @@ def test_retrieve_jobs_by_session_id(self, service):
         rjobs = service.jobs(session_id="test")
         self.assertFalse(rjobs)
 
+    @run_integration_test
+    def test_jobs_filter_by_date(self, service):
+        """Test retrieving jobs by creation date."""
+        current_time = datetime.now(timezone.utc)
+        job = self._run_program(service)
+        job.wait_for_final_state()
+        time_after_job = datetime.now(timezone.utc)
+        rjobs = service.jobs(created_before=time_after_job, created_after=current_time)
+        self.assertEqual(1, len(rjobs), f"Retrieved jobs: {[j.job_id for j in rjobs]}")
```

Comment:
This is not failure proof, there could be other jobs started at that time by cron jobs on the test account, so I would check if the job we created is in the list of jobs returned.